services/collaboration/pkg/connector/fileconnector_test.go

```diff
@@ -864,9 +871,72 @@ var _ = Describe("FileConnector", func() {
 			// Check both separately
 			Expect(newFileInfo.UserId).To(HavePrefix(hex.EncodeToString([]byte("guest-"))))
 			Expect(newFileInfo.UserFriendlyName).To(HavePrefix("Guest "))
-			// overwrite UserId and UserFriendlyName here for easier matching
+			Expect(newFileInfo.WatermarkText).To(HavePrefix("Guest "))
+			// overwrite UserId, UserFriendlyName and WatermarkText here for easier matching
 			newFileInfo.UserId = "guest-zzz000"
 			newFileInfo.UserFriendlyName = "guest zzz000"
+			newFileInfo.WatermarkText = "guest zzz000"
+
+			Expect(err).To(Succeed())
+			Expect(newFileInfo).To(Equal(expectedFileInfo))
+		})
+
+		It("Stat success authenticated user", func() {
+			// change view mode to view only
+			wopiCtx.ViewMode = appproviderv1beta1.ViewMode_VIEW_MODE_VIEW_ONLY
+
+			ctx := middleware.WopiContextToCtx(testCtx, wopiCtx)
+
+			gatewayClient.On("Stat", mock.Anything, mock.Anything).Times(1).Return(&providerv1beta1.StatResponse{
+				Status: status.NewOK(ctx),
+				Info: &providerv1beta1.ResourceInfo{
+					Owner: &userv1beta1.UserId{
+						Idp:      "customIdp",
+						OpaqueId: "aabbcc",
+						Type:     userv1beta1.UserType_USER_TYPE_PRIMARY,
+					},
+					Size: uint64(998877),
+					Mtime: &typesv1beta1.Timestamp{
+						Seconds: uint64(16273849),
+					},
+					Path: "/path/to/test.txt",
+					// Other properties aren't used for now.
+				},
+			}, nil)
+
+			expectedFileInfo := connector.FileInfo{
+				OwnerId:                    "61616262636340637573746f6d496470", // hex of aabbcc@customIdp
+				Size:                       int64(998877),
+				Version:                    "16273849.0",
+				BaseFileName:               "test.txt",
+				BreadcrumbDocName:          "test.txt",
+				UserCanNotWriteRelative:    true,
+				HostViewUrl:                "http://test.ex.prv/view",
+				HostEditUrl:                "http://test.ex.prv/edit",
+				EnableOwnerTermination:     false,
+				SupportsExtendedLockLength: true,
+				SupportsGetLock:            true,
+				SupportsLocks:              true,
+				DisableExport:              true,
+				DisableCopy:                true,
+				DisablePrint:               true,
+				IsAnonymousUser:            false,
+				UserId:                     "guest-zzz000",
```

Comment:
This seems a bug to me... If we have an authenticated user, we should use his username, so the expectation is wrong.
I guess you've copied the test above, but it was a special case because of the random numbers of the guests. That's why there are comments for checking some specific values before overwriting them, but I don't think we need that piece of code because the expectation should be more predictable.

Reply:
Secure view is only for authenticated users.

Reply:
You are right. Changed for the authenticated users.